### ISSUE TITLE
Missing dependency of openslide: libjxr0

### ISSUE THREAD

**Issue by ylep** [brainvisa/casa-distro#296]
According to @nsouedet the OpenSlide library is not built with libjxr, which breaks CZI I/O.

Indeed `libjxr0` is not included in the `casa-run` images and it probably should be (see https://github.com/brainvisa/casa-distro/pull/295).

**Comment by ylep** [comment]
@nsouedet is adding `libjxr0` to the casa-run images enough to fix the issue (https://github.com/brainvisa/casa-distro/issues/296)?

On one hand, the `libjxr-dev` package is already installed during compilation of openslide,

https://github.com/brainvisa/casa-distro/blob/febe85389fc0710efa0c89ce643fda3a3f84467f/image-recipes/casa-run/5.0/build_dependencies.sh#L34

and it compiles just fine:

https://github.com/brainvisa/casa-distro/blob/febe85389fc0710efa0c89ce643fda3a3f84467f/image-recipes/casa-run/5.0/install_compiled_dependencies.sh#L50-L59

On the other hand, I see that Denis recently had to force the detection of that library for the Ubuntu 22-based image, do we need to implement this trick for the Ubuntu 18-based image as well?

https://github.com/brainvisa/casa-distro/blob/febe85389fc0710efa0c89ce643fda3a3f84467f/image-recipes/casa-run/5.3/install_compiled_dependencies.sh#L25-L26

**Comment by denisri** [comment]
I don't remember all the details but yes I had to do someting manually to add jxrlib in the ubuntu 22 image. Maybe the missing runtime was the key of the problem. We need a test image I think...

**Comment by ylep** [comment]
Seems that adding `libjxr0` to the casa-run-5.0 image is not enough:

```
$ AimsFileInfo toto.czi 
OpenSlide: JPEGXR compression not supported
OpenSlide: JPEGXR compression not supported
OpenSlide: JPEGXR compression not supported
OpenSlide: JPEGXR compression not supported
AimsFileInfo: /volatile/wip/2021-12_czi/CJ1301_BX107_NEUN_91.czi : invalid format (not handled)
```

We need to check how openslide is compiled, because it does not seem to detect that `libjxr-dev` is installed

**Comment by ylep** [comment]
Indeed the `configure` script does not detect `libjxr-dev` even though it is installed:

```
checking for LIBJXR... no
```

**Comment by denisri** [comment]
by the way libjxr-dev is also not installed in dev images, but this is quite independent since we install libjxr-dev transciently during installation of openslide in run images.

**Comment by ylep** [comment]
@nsouedet and I fixed the issue in #297, I am starting to rebuild the `casa-run-5.0` and `casa-dev-5.0` images based on that.

**Comment by denisri** [comment]
It doesn't build yet...

**Comment by ylep** [comment]
@denisri I know, I had to stop in the middle of debugging this issue... Thanks for spotting the `libjxr`/`jxrlib` mistake.

There seem to be two different build systems for libjxr (a plain `Makefile` and CMake), which install to different locations, and to further complicate matters, the upstream pkg-config file corresponds to the wrong location... :expressionless: I will report that in the Debian bug tracker

**Comment by ylep** [comment]
Fixed in `casa-run-5.0-5.sif` and `casa-dev-5.0-8.sif`
